go/roothash/api/api.go

```diff
@@ -217,6 +217,9 @@ type SubmitMsg struct {
 	// Tokens are any tokens sent into the runtime as part of the message being sent. The tokens are
 	// transferred before the message is processed by the runtime.
 	Tokens quantity.Quantity `json:"tokens,omitempty"`
+	// Transaction is an optional transaction. `nil` means no transaction, while `tx = nil; &tx` is
+	// a transaction that is the empty byte string. Go, I swear.
+	Transaction *[]byte `json:"transaction,omitempty"`
```

Comment:
This breaks consensus, why is this new field needed? Any message data is runtime-dependent and should go into `Data`. It's not like only some messages (with this transaction field set?) should be propagated into runtimes, all incoming messages should be.

Reply:
not needed, although it saves us the overhead of adding an extra entry into the runtime to parse out the transactions.

it's also not this field alone that breaks consensus, is it? or would we not enter the other changes as breaking?

Reply:
hm I guess that would only affect the runtime committee.

let me try having the .Data be a serialized structure that contains an optional transaction. we can have the node understand that, and still avoid an extra entry into the runtime